tests/std/tests/P0466R5_layout_compatibility_and_pointer_interconvertibility_traits/test.cpp

```diff
@@ -98,14 +122,21 @@ constexpr bool test() {
         ASSERT(is_pointer_interconvertible_base_of_v<A, volatile C>);
         ASSERT(is_pointer_interconvertible_base_of_v<volatile A, const C>);
         ASSERT(is_pointer_interconvertible_base_of_v<A, D>);
-        ASSERT(is_pointer_interconvertible_base_of_v<A, A1>);
-        ASSERT(is_pointer_interconvertible_base_of_v<A, A2>);
-        ASSERT(is_pointer_interconvertible_base_of_v<A, A3>);
-        ASSERT(is_pointer_interconvertible_base_of_v<A2, A3>);
-
-        ASSERT(!is_pointer_interconvertible_base_of_v<A1, A2>);
-        ASSERT(!is_pointer_interconvertible_base_of_v<A4, A>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, E>);
+        ASSERT(is_pointer_interconvertible_base_of_v<A, F>);
+        ASSERT(is_pointer_interconvertible_base_of_v<C, D>);
+        ASSERT(is_pointer_interconvertible_base_of_v<I, I>);
+        ASSERT(is_pointer_interconvertible_base_of_v<const I, I>);
+
+        ASSERT(!is_pointer_interconvertible_base_of_v<int, int>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<void, void>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<A, int>);
         ASSERT(!is_pointer_interconvertible_base_of_v<B, C>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<A, NS>);
+        ASSERT(!is_pointer_interconvertible_base_of_v<B, NS>);

```

Comment:
This is an extremely good test (I noticed that it goes beyond my previous round of feedback, and tests the standard-layout part of pointer-interconvertible, arriving at a different answer than classic `is_base_of_v`). Thanks for adding it!